CMakeLists.txt

```diff
@@ -73,7 +73,12 @@ string(TOLOWER ${ONEDPL_BACKEND} ONEDPL_BACKEND)
 
 if (ONEDPL_ENABLE_SIMD)
     # TODO: remove skipping of Intel® oneAPI DPC++ Compiler once it can handle such options.
-    if (NOT CMAKE_CXX_COMPILER MATCHES ".*dpcpp(-cl)?(.exe)?$")
+    if (CMAKE_CXX_COMPILER MATCHES ".*dpcpp(-cl)?(.exe)?$")
+        execute_process(COMMAND dpcpp --version OUTPUT_VARIABLE VER)
```

Comment:
Two points:
1. Use of `CMAKE_CXX_COMPILER` instead of `dpcpp` looks more correctly but
2. Need to make sure it works fine on both Linux and Windows.

Reply:
1. Applied
2. Tested